scripts/monte_carlo.py

```diff
@@ -147,20 +189,161 @@ def monte_carlo_sampling_scipy(
     Documentation for Latin Hypercube: https://docs.scipy.org/doc/scipy/reference/generated/scipy.stats.qmc.LatinHypercube.html#scipy.stats.qmc.LatinHypercube
     Orthogonal LHS is better than basic LHS: https://github.com/scipy/scipy/pull/14546/files, https://en.wikipedia.org/wiki/Latin_hypercube_sampling
     """
+    from scipy.stats import qmc
+
     sampler = qmc.LatinHypercube(
         d=N_FEATURES,
         centered=centered,
         strength=strength,
         optimization=optimization,
         seed=seed,
     )
+
     lh = sampler.random(n=SAMPLES)
+
+    lh = rescale_distribution(lh, uncertainties_values)
     discrepancy = qmc.discrepancy(lh)
-    logger.info(f"Hypercube discrepancy is: {discrepancy}")
+    logger.info(
+        "Discrepancy is:", discrepancy, " more details in function documentation."
+    )
 
     return lh
 
 
+def rescale_distribution(latin_hypercube, uncertainties_values):
+    """
+    Rescales a Latin hypercube sampling (LHS) using specified distribution
+    parameters.
+
+    Parameters:
+    - latin_hypercube (np.array): The Latin hypercube sampling to be rescaled.
+    - uncertainties_values (list): List of dictionaries containing distribution information.
+        Each dictionary should have 'type' key specifying the distribution type and 'args' key
+        containing parameters specific to the chosen distribution.
+
+    Returns:
+    - np.array: Rescaled Latin hypercube sampling with values in the range [0, 1].
+
+    Supported Distributions:
+    - "Uniform": No rescaling applied.
+    - "Normal": Rescaled using the inverse of the normal distribution function with specified mean and std.
+    - "LogNormal": Rescaled using the inverse of the log-normal distribution function with specified mean and std.
+    - "Triangle": Rescaled using the inverse of the triangular distribution function with mean calculated from given parameters.
+    - "Beta": Rescaled using the inverse of the beta distribution function with specified shape parameters.
+    - "Gamma": Rescaled using the inverse of the gamma distribution function with specified shape and scale parameters.
+
+    Note:
+    - The function supports rescaling for Uniform, Normal, LogNormal, Triangle, Beta, and Gamma distributions.
+    - The rescaled samples will have values in the range [0, 1].
+    """
+    from scipy.stats import beta, gamma, lognorm, norm, triang
+    from sklearn.preprocessing import MinMaxScaler
+
+    for idx, value in enumerate(uncertainties_values):
+        dist = value.get("type")
+        params = value.get("args")
+
+        match dist:
+            case "Uniform":
+                pass
+            case "Normal":
+                mean, std = params
+                latin_hypercube[:, idx] = norm.ppf(latin_hypercube[:, idx], mean, std)
+            case "LogNormal":
+                mean, std = params
+                latin_hypercube[:, idx] = lognorm.ppf(latin_hypercube[:, idx], s=0.90)
```

Comment:
what is s=0.9?
should that be another parameter?

Reply:
Yeah. I should pass it through the parameters var

Reply:
yeah, isn't it "shape"?

Reply:
You are correct. It is shape.